parser.cc

```diff
@@ -1453,6 +1454,52 @@ Input::Index AsciiFileInput::size() const
 	return file_size;
 }
 
+StreamInput StreamInput::Create(std::istream& s)
+{
+	const size_t start = s.tellg();
+	s.seekg(0, std::ios::end);
+
+	const size_t len = static_cast<size_t>(s.tellg()) - start;
+	s.seekg(start, std::ios::beg);
```

Comment:
That's what I'm thinking, yes. I'm not sure I can think of a good reason to consume only part of a file, but there's very little extra effort required to support the most general case.